controllers/operatorpolicy_controller.go

```diff
@@ -108,167 +101,300 @@ func (r *OperatorPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	// Start query batch for caching and watching related objects
 	err = r.DynamicWatcher.StartQueryBatch(watcher)
 	if err != nil {
-		OpLog.Error(err, "Could not start query batch for the watcher", "watcher", watcher.Name,
-			"watcherKind", watcher.Kind)
+		OpLog.Error(err, "Could not start query batch for the watcher")
 
 		return reconcile.Result{}, err
 	}
 
 	defer func() {
 		err := r.DynamicWatcher.EndQueryBatch(watcher)
 		if err != nil {
-			OpLog.Error(err, "Could not end query batch for the watcher", "watcher", watcher.Name,
-				"watcherKind", watcher.Kind)
+			OpLog.Error(err, "Could not end query batch for the watcher")
 		}
 	}()
 
 	// handle the policy
-	OpLog.Info("Reconciling OperatorPolicy", "policy", policy.Name)
+	OpLog.Info("Reconciling OperatorPolicy")
 
-	// Check if specified Subscription exists
-	cachedSubscription, err := r.DynamicWatcher.Get(watcher, subscriptionGVK, policy.Spec.Subscription.Namespace,
-		policy.Spec.Subscription.SubscriptionSpec.Package)
-	if err != nil {
-		OpLog.Error(err, "Could not get subscription", "kind", subscriptionGVK.Kind,
-			"name", policy.Spec.Subscription.SubscriptionSpec.Package,
-			"namespace", policy.Spec.Subscription.Namespace)
+	if err := r.handleOpGroup(ctx, policy); err != nil {
+		OpLog.Error(err, "Error handling OperatorGroup")
 
 		return reconcile.Result{}, err
 	}
 
-	subExists := cachedSubscription != nil
+	if err := r.handleSubscription(ctx, policy); err != nil {
+		OpLog.Error(err, "Error handling Subscription")
 
-	// Check if any OperatorGroups exist in the namespace
-	ogNamespace := policy.Spec.Subscription.Namespace
-	if policy.Spec.OperatorGroup != nil {
-		ogNamespace = policy.Spec.OperatorGroup.Namespace
+		return reconcile.Result{}, err
 	}
 
-	cachedOperatorGroups, err := r.DynamicWatcher.List(watcher, operatorGroupGVK, ogNamespace, nil)
+	// FUTURE: more resource checks
+
+	return reconcile.Result{}, nil
+}
+
+func (r *OperatorPolicyReconciler) handleOpGroup(ctx context.Context, policy *policyv1beta1.OperatorPolicy) error {
+	watcher := opPolIdentifier(policy.Namespace, policy.Name)
+
+	desiredOpGroup, err := buildOperatorGroup(policy)
 	if err != nil {
-		OpLog.Error(err, "Could not list operator group", "kind", operatorGroupGVK.Kind,
-			"namespace", ogNamespace)
+		return fmt.Errorf("error building operator group: %w", err)
+	}
 
-		return reconcile.Result{}, err
+	foundOpGroups, err := r.DynamicWatcher.List(
+		watcher, operatorGroupGVK, desiredOpGroup.Namespace, labels.Everything())
```

Comment:
Is `labels.Everything()` basically no selector? If so, you can optionally provide `nil` instead for the label selector.